File: 590-N-ary_Tree_Postorder_Traversal/code.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution {
public:
    vector<int> postorder(Node* root) {
        if (root == NULL){
            return {};
        }
        
        vector<int> res;
        stack<Node*> st;
        st.push(root);
        while(!st.empty()){
            Node* tmp = st.top();
            st.pop();
            for (int i = 0; i < tmp->children.size(); i++){
                st.push(tmp->children[i]);
            }
            res.push_back(tmp->val);
        }
        reverse(res.begin(), res.end());
        return res;
        
        
    }
};

File: 589-N_ary_Tree_Preorder_Traversal/code.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""
class Solution(object):
    def preorder(self, root):
        """
        :type root: Node
        :rtype: List[int]
        """
        def dfs(root):
            if not root:
                return None
            res.append(root.val)
            for child in root.children:
                dfs(child)
                
        res = []
        dfs(root)
        return res